packages/core/src/amazonqGumby/commands.ts

```diff
@@ -5,28 +5,41 @@
 
 import * as vscode from 'vscode'
 import { telemetry } from '../shared/telemetry/telemetry'
-import { ExtContext } from '../shared/extensions'
 import { Commands } from '../shared/vscode/commands2'
 import * as CodeWhispererConstants from '../codewhisperer/models/constants'
-import { startTransformByQWithProgress, confirmStopTransformByQ } from '../codewhisperer/commands/startTransformByQ'
+import { stopTransformByQ } from '../codewhisperer/commands/startTransformByQ'
 import { transformByQState } from '../codewhisperer/models/model'
 import { AuthUtil } from '../codewhisperer/util/authUtil'
 import { CancelActionPositions, logCodeTransformInitiatedMetric } from './telemetry/codeTransformTelemetry'
+import { ChatControllerEventEmitters } from './chat/controller/controller'
+import { focusAmazonQPanel } from '../auth/ui/vue/show'
+import { sleep } from '../shared/utilities/timeoutUtils'
+import { randomUUID } from 'crypto'
 
 export const showTransformByQ = Commands.declare(
     { id: 'aws.awsq.transform', compositeKey: { 0: 'source' } },
-    (context: ExtContext) => async (source: string) => {
+    (controllerEventEmitters: ChatControllerEventEmitters) => async (source: string) => {
         if (AuthUtil.instance.isConnectionExpired()) {
             await AuthUtil.instance.notifyReauthenticate()
         }
 
         if (transformByQState.isNotStarted()) {
             logCodeTransformInitiatedMetric(source)
-            await startTransformByQWithProgress()
+            await focusAmazonQPanel().then(async () => {
+                // non blocking wait for a tenth of a second
+                // note: we need this because the webview has to be loaded
+                // before it can listen to events we fire
+                await sleep(100)
```

Comment:
Learning: There is some data from Nielsons Heuristics and 100-200ms is perceived as almost instant for humans. 300-500ms is an "easing" motion. This principles are for UX animations, but this is similar.

<img width="675" alt="Screenshot 2024-03-14 at 10 58 38 AM" src="https://github.com/aws/aws-toolkit-vscode/assets/38665452/869a5cc9-a222-4e4d-84d5-c8747affdf22">

Reply:
Oh interesting, thank you for providing this!